processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -172,10 +185,56 @@ private static ColumnValueSelector getColumnValueSelectorFromDimensionSpec(
   private static <ColumnSelectorStrategyClass extends ColumnSelectorStrategy> ColumnSelectorStrategyClass makeStrategy(
       ColumnSelectorStrategyFactory<ColumnSelectorStrategyClass> strategyFactory,
       String dimName,
-      ColumnCapabilities capabilities
+      ColumnCapabilities capabilities,
+      boolean hasExFn
   )
   {
-    capabilities = getEffectiveCapabilities(dimName, capabilities);
+    capabilities = getEffectiveCapabilities(dimName, capabilities, hasExFn);
     return strategyFactory.makeColumnSelectorStrategy(capabilities);
   }
+
+  private static final Pattern LONG_PAT = Pattern.compile("[-|+]?\\d+");
+
+  // TopNBinaryFn relies on this returning the same Long object when no conversion is needed
+  public static Long convertObjectToLong(Object valObj)
+  {
+    if (valObj instanceof Long) {
+      return (Long) valObj;
+    } else if (valObj instanceof Number) {
+      return ((Number) valObj).longValue();
+    } else if (valObj instanceof String) {
+      try {
+        String s = ((String) valObj).replace(",", "");
```

Comment:
Well, I suppose the purpose is to remove 1000s delimiter from numbers, e. g. `1,000,000` is `1000000`. But not in all locales, e. g. in Russian locale comma is a separator between integer part and fractional part in non-integer numbers.

IMO this is "too smart" and we shouldn't do this. Probably it should be discussed separately, not in this PR